Refactor kube filter tests and consequently fix some highlighted bugs (1/2)

yann-soubeyrand opened the pull request.

edsiper commented:
hi @yann-soubeyrand 

please explain/list the changes made on this PR to facilitate it review

yann-soubeyrand commented:
Hi @edsiper,

- I moved test logs, meta and out files in distinct folders to better organize them.
- I removed the Docker ID from log file names as they aren't needed to test Kubernetes filter plugin (they test tail input plugin). This leads to much shorter file names.
- Doing so, uncovered a bug in the code which gets Docker ID and container image ID from the API server, which I fixed.
- I added some error checking in the test runner to fail more gracefully.
- I cleaned test runner output, trying to have just the test name and the result next to it (there is however still a line of log I couldn't remove).
- Doing so, I conditioned the code gathering info from the API server, to only execute when Fluent Bit is running in a pod. This is to suppress an error output when running the tests.

Overall, the goal of this PR is to make test runner more robust, with a cleaner output and to make it easier to manage tests. There is no change in the tests nor in the test runner mechanism.

The diff is horrible and a real pain to review because the test files are so small and with changes that Git does not see them as renamed but as deleted and with new files created. There is nothing I can do for this fact however :-(

edsiper commented:
thanks!
